extensions/compose/src/detect.spec.ts

```diff
@@ -228,19 +226,19 @@ describe('Check docker socket', async () => {
       };
     });
 
-    const spyGet = vi.spyOn(http, 'get') as unknown as MockInstance;
+    const spyGet = vi.spyOn(http, 'get') as MockInstance;
```

Comment:
By doing this, we are losing the type of the spy.

Instead, I would do:

(it seems we are forced to cast the signature of `http.get` to the one we want to use, as `vi.mocked` is not able to determine which signature to use, and fallbacks to the wrong one)
```
// this one may be moved to the top of the file, as it is hoisted anyway (https://vitest.dev/api/vi#vi-mock).
vi.mock('node:http');

const clientRequestEmitter = new EventEmitter();
const myRequest = clientRequestEmitter as http.ClientRequest;

vi.mocked(
  http.get as (
    options: http.RequestOptions | string | URL,
    callback: (res: http.IncomingMessage) => void,
  ) => http.ClientRequest,
).mockImplementation((_options, callback) => {
[...]
```

Reply:
Fixed by using similar approach.